### ISSUE TITLE
Provide guidelines for contributing regression tests

### ISSUE THREAD

**Issue by t0yv0** [pulumi/pulumi-terraform-bridge#2203]
Per @corymhall feedback it would be very good to centralize the flow of turning bridged provider tests into bridge regression test and adding them. We start with a provider such as pulumi-aws, a Pulumi program, and a scenario, and we need to turn this into a minimal repro that mimicks the outer provider definition and reproduces the issue (with bonus points such as making debugger attachable or x-testing this vs TF CLI). There's now a couple of ways to do this which can be confusing for contributors.

**Comment by iwahbe** [comment]
I agree that this would be a good idea.

**Comment by VenelinMartinov** [comment]
I'd encourage us to move away from the GRPC regression tests as these can be quite sensitive and overfit on bridge internals.

I've got a lot of mileage out of the schema + yaml program + pulumitest tests in the SDK: https://github.com/pulumi/pulumi-terraform-bridge/blob/f317dc4a503d3430fbe13f3af964207f0a17616e/pkg/tests/schema_pulumi_test.go#L19

We now have similar tests in PF: https://github.com/pulumi/pulumi-terraform-bridge/blob/67c3b5765ae629a67e9ab69815c09a8b3a015098/pf/tests/schema_and_program_test.go#L18

Is this a reasonable place to point new contributors to?

**Comment by t0yv0** [comment]
Yes indeed, gRPC-level tests are not the right form for the common regression testing use case, we abused them a bit. These tests you point to are about ideal for the purpose. That's a great starting point.

**Comment by VenelinMartinov** [comment]
We have this guide now: https://pulumi-developer-docs.readthedocs.io/projects/pulumi-terraform-bridge/en/latest/docs/guides/testing.html
